src/trajectory/util.cpp

```diff
@@ -28,38 +36,16 @@ namespace trajectory {
 
 namespace {
 
-bool checkStateSpace(const statespace::StateSpace* _stateSpace)
+enum AllowJointType
 {
-  // TODO(JS): Generalize Rn<N> for arbitrary N.
-  if (dynamic_cast<const R<0>*>(_stateSpace) != nullptr)
-  {
-    return true;
-  }
-  else if (dynamic_cast<const R<1>*>(_stateSpace) != nullptr)
-  {
-    return true;
-  }
-  else if (dynamic_cast<const R<2>*>(_stateSpace) != nullptr)
-  {
-    return true;
-  }
-  else if (dynamic_cast<const R<3>*>(_stateSpace) != nullptr)
-  {
-    return true;
-  }
-  else if (dynamic_cast<const R<4>*>(_stateSpace) != nullptr)
-  {
-    return true;
-  }
-  else if (dynamic_cast<const R<5>*>(_stateSpace) != nullptr)
-  {
-    return true;
-  }
-  else if (dynamic_cast<const R<6>*>(_stateSpace) != nullptr)
-  {
-    return true;
-  }
-  else if (dynamic_cast<const R<Eigen::Dynamic>*>(_stateSpace) != nullptr)
+  SinglularDOF,
+  NonSingularDOF
+};
+
+bool checkStateSpace(const statespace::StateSpace* _stateSpace, AllowJointType type)
```

Comment:
@brianhou @gilwoolee :

There are a couple of tests written (planner / trajectory namespaces for example) that instantiate an R2 statespace and use this function during the process. I agree that we should not have two functions doing the same here. This seems to be a reasonable solution for now, especially given we currently have support for multi-DOF joints some places and not others. I would like to have this solution in place with an issue raised to
- Change the tests to be cartesian product spaces with multiple R1 joints.
or
- Extend support for multi-DOF joint spaces uniformly and remove this enum usage.

Since this method is internal, I am okay with using an enum at this point but would liek to get rid of it soon. What do you guys think?

Reply:
I would prefer to remove the enum and just use a Cartesian product space of multiple R1 joints in these tests, as part of this PR. Then, we can revisit the multi-DOF joint space if/when it becomes useful in the future.

Reply:
I agree with Brian. Can we have an issue to track this in case it becomes necessary?